router/http/src/main/scala/io/buoyant/router/http/ForwardClientCertFilter.scala

```diff
@@ -0,0 +1,66 @@
+package io.buoyant.router.http
+
+import java.security.MessageDigest
+import java.security.cert.X509Certificate
+import javax.xml.bind.DatatypeConverter.printHexBinary
+
+import com.twitter.finagle._
+import com.twitter.finagle.http.{Request, Response}
+import com.twitter.finagle.transport.Transport
+import com.twitter.util.Future
+
+import scala.collection.mutable
+
+class ForwardClientCertFilter extends SimpleFilter[Request, Response] {
```

Comment:
Can you factor out common code between this an the H2 version to reduce code duplication?